tests/tamr_client/datasets/test_record.py

```diff
@@ -0,0 +1,139 @@
+from functools import partial
+import json
+from typing import Dict
+
+import pytest
+import responses
+
+import tamr_client as tc
+import tests.tamr_client.utils as utils
+
+
+@responses.activate
+def test_update():
+    s = utils.session()
+    dataset = utils.dataset()
+
+    url = tc.URL(path="datasets/1:updateRecords")
+    updates = records_to_updates(_records_json)
+    snoop: Dict = {}
+    responses.add_callback(
+        responses.POST, url.__str__(), partial(create_callback, snoop=snoop, status=200)
```

Comment:
The `__str__` method (btw `__<name>__ is pronounced "dunder <name>") should not be called directly. Rather, it allows [the built-in `str` function to do the right thing](https://docs.python.org/3/library/stdtypes.html#str):

```suggestion
        responses.POST, str(url), partial(create_callback, snoop=snoop, status=200)
```